torch/utils/data/datapipes/iter/combinatorics.py

```diff
@@ -116,12 +116,15 @@ def set_shuffle(self, shuffle=True):
 
     def set_seed(self, seed: int):
         self._seed = seed
+        return self
 
     def __iter__(self) -> Iterator[T_co]:
         if not self._enabled:
             for x in self.datapipe:
                 yield x
         else:
+            if self._seed is None:
+                self._seed = int(torch.empty((), dtype=torch.int64).random_().item())
```

Comment:
@NivekT There is a separate problem for `reset`.

As I said about `reset`, it is not invoked when the datapipe state is `NotStarted`. So, I have to add the manual reset logic here to generate `seed`. But, it prevents this DataPipe  not resetting after deserialization unless I add an extra flag to indicate if the datapipe is restored or not (this is contrary to our idea to use `reset` by default).

Reply:
I we do so, then I will remove the manual reset logic here.

Reply:
I think the extra reset may be somewhat unnecessary for most DataPipes in the beginning. However, I don't think it should be noticeable and should not break anything. I just tried looking through if it will causes any edge cases for cases with ChildDataPipes but I can't think of any. You can make that change and confirm.